### PR TITLE
Add speed color legend

### PR DESCRIPTION
**Pull request by jcardus**
Screenshot 2024-11-04 at 17 00 47

### CODE REVIEW

**Comment by jcardus** [comment]
Screenshot 2024-11-04 at 16 50 07

**Comment by tananaev** [comment]
I think this is overkill. I don't want to add more controls just for speed.

**Comment by jcardus** [comment]
It goes out when we leave route replay

**Comment by tananaev** [comment]
I understand, but still it takes a lot of space.

**Comment by jcardus** [comment]
any suggestion?

**Comment by tananaev** [comment]
I think more reasonable would be to maybe show some markers on the map to indicate speed. Maybe show it on hover only.

**Comment by jcardus** [comment]
This way you get a quick idea of the speeding (where and how much). Maybe I can put it smaller.

**Comment by jcardus** [comment]
Smaller version...


localhost_3000_(iPhone 12 Pro) (1)

**Comment by tananaev** [comment]
Looks better. Can we make text the same style as the distance control? We can also probably round to 1 or even 0 digits. And maybe we don't need the middle point. Then we can reduce the width even more.

**Comment by jcardus** [comment]
here it goes
localhost_3000_(iPhone 12 Pro) (1)

File: src/map/legend/MapSpeedLegend.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    const legend = document.createElement('div');
+
+    const colorBar = document.createElement('div');
+    colorBar.style.background = `linear-gradient(to right, ${gradientStops})`;
+    colorBar.style.height = '10px';
+
+    const speedLabel = document.createElement('span');
+    speedLabel.textContent = `${Math.round(speedFromKnots(this.minSpeed, this.speedUnit))} - ${
+      Math.round(speedFromKnots(this.maxSpeed, this.speedUnit))} ${speedUnitString(this.speedUnit, this.t)}`;
```
Comment (tananaev) [inline src/map/legend/MapSpeedLegend.js:47]: Maybe split these calculations into separate variables for readability.

File: src/map/legend/MapSpeedLegend.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { interpolateTurbo } from '../../common/util/colors';
+import { speedFromKnots, speedUnitString } from '../../common/util/converter';
+
+export class LegendControl {
```
Comment (tananaev) [inline src/map/legend/MapSpeedLegend.js:4]: Probably better to call it `SpeedLegendControl`.

**Comment by tananaev** [comment]
Looks good, thank you.

**Comment by hypoclone** [comment]
Hello @jcardus and @tananaev, thank you guys for the great work you do. i don't know if it is ok to discuss it here or create another discussion on it, but i think it will be intuitive to have the replay track show stops and stops duration with an icon i.e

https://github.com/user-attachments/assets/b23b34df-78dd-484a-b26e-7fdd75c3a1f2

**Comment by jcardus** [comment]
Maybe it's better to create a feature request, probably for the combined report which already has some of that information.

**Pull request merged**